src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -213,9 +213,6 @@ def _install_deps_for_psycopg2():  # pylint: disable=too-many-statements
         distname, _ = get_linux_distro()
         distname = distname.lower().strip()
         if installer == 'DEB' or any(x in distname for x in ['ubuntu', 'debian']):
-            from azure.cli.core.util import in_cloud_console
-            if in_cloud_console():
-                raise CLIError("This extension is not supported in Cloud Shell as you do not have permission to install extra dependencies.")
```

Comment:
I think we should skip following steps for Cloud Shell since those dependencies will be installed by https://github.com/Azure/CloudShell/pull/143/files.

Reply:
I mean, why do we bother with the detection since they will always be available after https://github.com/Azure/CloudShell/pull/143, right?

Reply:
Yes for CloudShell, your saying is correct. But my understanding is that starting from elif system == 'Linux': (212)  these are for general Linux system and not just targeting CloudShell. If a customer is using a Linux based OS, we still want to install these for them

Reply:
I mean, skip these lines if cloud shell is detected.